src/providers/user.ts

```diff
@@ -66,6 +66,11 @@ export class User {
     return this.beeminder.fetchDatapoints(goal.slug);
   }
 
+  addDataPoint(goal, datapoint){
+    this.beeminder.addDataPoint(goal, datapoint).subscribe(() => 
+      this.goals.push(goal));
```

Comment:
You don't really need to re push the goal object, because our goals object doesn't hold datapoint.